src/parsing/Parse_target.ml

```diff
@@ -111,8 +119,23 @@ let parse_and_resolve_name lang file =
   AST_generic.SId.unsafe_reset_counter ();
   Naming_AST.resolve lang ast;
   Typing.check_program lang ast;
+
+  let cfgs = CFG_build.cfgs_in_program lang ast in
+
+  (* Flow-insensitive constant propagation. *)
   Constant_propagation.propagate_basic lang ast;
-  Constant_propagation.propagate_dataflow lang ast;
+
+  (* Flow-sensitive constant propagation. *)
+  cfgs
+  |> List.iter (fun (inputs, flow) ->
```

Comment:
`Constant_propagation` and `Implicit_return` could just take a `program_cfgs` type and do the iteration, so here you would just have a call to `Constant_propagation.propagate_dataflow_progam cfgs` or something.

Reply:
Done.